src/handlers/handle-transfers.ts

```diff
@@ -0,0 +1,105 @@
+import { log, Address, BigInt, Bytes } from "@graphprotocol/graph-ts";
+import { Transfer as TransferEvent } from "../../generated/SingleEditionMintableCreator/SingleEditionMintable"
+import { FreeNFTBalance, FreeNFTDrop } from "../../generated/schema";
+
+
+let NULL_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+
+function updateBalance(
+  event: TransferEvent, 
+  drop: FreeNFTDrop, 
+  address: Address, 
+  isSender: boolean = false,
+): void {
```

Comment:
consider this a mega-nit if you will but
- I'm not a big fan of the `isSender` bool to change behavior
- most of `updateBalance` is concerned with actually loading/creating a balance, I think it makes sense to break it out

I would do it like this:

```ts
function loadBalance(
  drop: FreeNFTDrop,
  address: Address,
): FreeNFTBalance | null {
  if (address.toHexString() == NULL_ADDRESS) {
    // we don't track balances for the null address
    return null;
  }

  let id = drop.id.toHexString() + "-" + address.toHexString();
  let balance = FreeNFTBalance.load(id);

  if (balance == null) {
    balance = new FreeNFTBalance(id);

    balance.address = address;
    balance.quantity = BigInt.zero();
    balance.drop = drop.id;
  }

  if (balance.address != address) {
    log.error(
      "Balance with id {} has owner address {}, which does not match address {}",
      [
        balance.id,
        balance.address.toHexString(),
        address.toHexString(),
      ]
    );

    return null;
  }

  return balance;
}
```

```ts
function updateBalance(
  balance: FreeNFTBalance | null,
  quantityDelta: i32,
  event: TransferEvent,
): void {
  if (balance == null) {
    return;
  }

  balance.quantity = balance.quantity.plus(BigInt.fromI32(quantityDelta));

  if (balance.quantity.lt(BigInt.zero())) {
    log.error(
      "Balance with id {} has negative quantity {} after transfer in tx {}",
      [
        balance.id,
        balance.quantity.toString(),
        event.transaction.hash.toHexString(),
      ],
    );

    return;
  }

  balance.updatedAt = event.block.timestamp;
  balance.save();
}
```

In `handleTransfer`:

```
  updateBalance(loadBalance(drop, sender), -1, event);
  updateBalance(loadBalance(drop, receiver), 1, event);
```

Reply:
100% agree, this is much, much nicer! making the change!